### ISSUE TITLE
InstantMessaging/Presence support

### ISSUE THREAD

**Issue by barbosas** [voximplant/react-native-voximplant#21]
Hello. I wonder if support for InstantMessaging / Presence support has already been implemented and how should I proceed to use

**Comment by aylarov** [comment]
@barbosas it hasn't been implemented yet, but we will update the SDK in the next month or so and it will probably include this functionality

**Comment by barbosas** [comment]
Hi @aylarov, do you have news?

**Comment by ir4y** [comment]
Is there any progress with adding InstantMessaging?

**Comment by YuliaGrigorieva** [comment]
Hello!

We are currently working on messaging support in the Voximplant React Native SDK. We plan to release this functionality in October.

Best regards,
Yulia Grigorieva

**Comment by jameslporter** [comment]
This is a roadblock for me as well. Only a week left in October! Any chance we'll see it this week? I've already tried out the messaging branch on android and emailed my fixes to support but I'll drop it in here too for easy reference. Just three missing imports in one file. Looks like it's the javascript side that's not completed or not pushed to the branch yet. I will gladly test anything you got even if it's a WIP.

import com.facebook.react.bridge.WritableMap;
import com.facebook.react.modules.core.DeviceEventManagerModule;
import javax.annotation.Nullable;

In the new file android/src/main/java/com/voximplant/reactnative/VIMessagingModule.java

**Comment by ruscoder** [comment]
Hello! Do you have any updates on this issue? Is the branch `messaging` under development?

**Comment by ruscoder** [comment]
Also, I've found a problem with `distinct` flag. I've run the tests and got this error:
```
expected Conversation {
  createdAt: 1548669425,
  uuid: '***',
  lastRead: 0,
  participants: Array [
    Object {
      canWrite: true,
      canManageParticipants: true,
      userId: '****'
    },
    Object {
      canWrite: true,
      canManageParticipants: false,
      userId: '***'
    },
    Object {
      canWrite: true,
      canManageParticipants: true,
      userId: '***'
    }
  ],
  isUber: false,
  lastSeq: 1,
  publicJoin: true,
  distinct: false,
  lastUpdate: 1548669425,
  customData: Object {},
  title: 'Test conversation 1'
} to have property distinct of true (got false)
```

**Comment by TreeMan360** [comment]
This is a major blocker for me using your service. When do you plan to finally complete merge the messaging branch into master?

**Comment by YuliaGrigorieva** [comment]
Hello!

Today we have released Voximplant React Native SDK 1.6.0 with Messaging API.

'messaging' branch will be completely removed from the repo on May 6th, 2019.

Best regards,
Yulia Grigorieva
